src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -72,7 +72,7 @@ public TransactionalState(ITransactionalStateConfiguration transactionalStateCon
         /// </summary>
         public void Save()
         {
-            var info = TransactionContext.GetTransactionInfo();
+            var info = (TransactionInfo) TransactionContext.GetTransactionInfo();
```

Comment:
As with other areas where we hard cast, avoid if we can, and if we can't avoid, we should throw meaningful exceptions.